Libraries/Text/RCTTextField.m

```diff
@@ -66,6 +66,28 @@ - (void)paste:(id)sender
   [super paste:sender];
 }
 
+- (void)setSelection:(NSDictionary *)selection
+{
+  NSInteger selectionStart = [RCTConvert NSInteger:selection[@"start"]];
+  NSInteger selectionEnd = [RCTConvert NSInteger:selection[@"end"]];
+  UITextRange *currentSelection = self.selectedTextRange;
+  NSInteger currentSelectionStart = [self offsetFromPosition:self.beginningOfDocument toPosition:currentSelection.start];
+  NSInteger currentSelectionEnd = [self offsetFromPosition:self.beginningOfDocument toPosition:currentSelection.end];
+  bool isSameSelection = (selectionStart == currentSelectionStart) && (selectionEnd == currentSelectionEnd);
+  NSInteger eventLag = _nativeEventCount - _mostRecentEventCount;
+  if (eventLag == 0) {
+    if (!isSameSelection) {
+      UITextPosition *start = [self positionFromPosition:[self beginningOfDocument] offset:selectionStart];
```

Comment:
sorry for nitpicking here, 
but you should use dot notion 'self.beginningOfDocument' in all places